rosbag2_transport/test/rosbag2_transport/test_record_all.cpp

```diff
@@ -104,23 +116,34 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_services_a
 
   start_async_spin(recorder);
 
+  // Wait until recorder discovery is complete, otherwise messages might be missed.
+  // The currently expected topics:
+  // /test_service_1/_service_event
+  // /test_service_2/_service_event
+  auto discovery_ret = rosbag2_test_common::wait_until_condition(
+    [&recorder]() {
+      return recorder->subscriptions().size() == 2;
+    },
+    std::chrono::seconds(5));
+  ASSERT_TRUE(discovery_ret);
```

Comment:
i believe that waiting on this service event subscription discovery makes sense. Although services are discovered and matched, maybe service event subscriptions are not yet available.